cmd/convox/builds.go

```diff
@@ -342,40 +367,35 @@ func cmdBuildsImport(c *cli.Context) error {
 		return stdcli.ExitError(err)
 	}
 
-	buildInput := os.Stdin
+	if !stdcli.IsTerminal(os.Stdin) && c.String("file") == "" {
+		return stdcli.ExitError(fmt.Errorf("please pipe a file into this command or specify -f"))
+	}
 
-	if len(c.Args()) == 1 {
-		file, err := os.Open(c.Args()[0])
+	in := os.Stdin
+	if file := c.String("file"); file != "" {
+		fd, err := os.Open(file)
 		if err != nil {
 			return stdcli.ExitError(err)
 		}
-
-		buildInput = file
-	}
-
-	b, err := ioutil.ReadAll(buildInput)
-	if err != nil {
-		return stdcli.ExitError(fmt.Errorf("error reading build: %s", err))
+		defer fd.Close()
+		in = fd
 	}
 
-	w := os.Stdout
+	out := os.Stdout
 
 	if c.Bool("id") {
-		w = os.Stderr
+		out = os.Stderr
 	}
 
-	build, err := rackClient(c).ImportBuild(app, b, progressFunc(w))
+	build, err := rackClient(c).ImportBuild(app, in, progressFunc(out))
 	if err != nil {
 		return stdcli.ExitError(err)
 	}
 
-	if c.Bool("id") {
-		fmt.Print(build.Release)
+	fmt.Fprintf(out, "\nRelease: %s\n", build.Release)
 
-	} else {
-		fmt.Println()
-		fmt.Printf("Imported %s\n", build.Id)
-		fmt.Printf("To deploy this copy run `convox releases promote %s --app %s`\n", build.Release, app)
+	if c.Bool("id") {
+		fmt.Println(build.Release)
```

Comment:
Nitpicking as the end result is the same but for consistency we could use `fmt.Println(build.Release)` as well in `cmdBuildsCreate()` instead of `os.Stdout.Write([]byte(fmt.Sprintf("%s\n", build)))`. Same naming scheme for `out := os.Stdout` too. This isn't a big deal.